### ISSUE TITLE
Add reference fine-tuning recipe with optimized hyperparameters for Nemotron Nano v2 with LoRA

### ISSUE THREAD

**Issue by zredlined** [NVIDIA-NeMo/NeMo#14856]
**Is your feature request related to a problem? Please describe.**

Currently, fine-tuning Nemotron models (especially Nano v2 and Super v1.5) with LoRA in NeMo Framework requires significant manual hyperparameter tuning and lacks production-ready defaults.
Specific challenges:

* No reference hyperparameters for fine-tuning Nemotron models, requiring expensive hyperparameter sweeps
* Manual weight merges and tokenizer handling required
* No streamlined adapter export functionality
* Lack of public reference recipes demonstrating fine-tuning best practices on standard benchmarks

**Describe the solution you'd like**

Currently, fine-tuning Nemotron models (especially Nano v2 and Super v1.5) with LoRA in NeMo Framework requires significant manual hyperparameter tuning and lacks production-ready defaults.

Specific challenges:
* No reference hyperparameters for fine-tuning Nemotron Nano 9b V2 model that demonstrates task improvement on a benchmark, requiring expensive hyperparameter sweeps
* Manual weight merges and tokenizer handling required
* No streamlined adapter export functionality
* Lack of public reference recipes demonstrating fine-tuning best practices on standard benchmarks

**Describe the solution you'd like**
Provide a complete, public fine-tuning recipe for Nemotron Nano v2 with LoRA that includes:

1. Optimized hyperparameters (learning rate, LoRA rank, alpha, dropout, batch size, etc.) validated on a public benchmark
2. Complete training script with best practices for weight merging, tokenizer handling, and adapter export
3. Public benchmark demonstration showing measurable performance improvement on tasks like:
    * Text-to-SQL (e.g., BIRD benchmark)
    * Mathematical reasoning (e.g., MATH, GSM8K)
    * Code generation (e.g., HumanEval, MBPP)
    * General knowledge (e.g., MMLU)

**Documentation** covering:

 * Expected training time and hardware requirements
 * How to adapt the recipe for different data scales
 * Best practices for preserving model capabilities (e.g., reasoning, structured output generation) during fine-tuning
Provide a code snippet on how new APIs/changes would be used by others.

**Hardware and performance targets**:

* Should run on 8xH100 or less powerful hardware
* Complete fine-tuning in approximately 4-8 hours on example datasets
* Example datasets should be <500k records to demonstrate practical fine-tuning at scale

Example Usage:
```
# Download reference recipe
git clone https://github.com/NVIDIA/NeMo-Framework-Recipes
cd fine-tuning/nemotron-nano-v2-lora

# Run with default optimized params
python finetune_nano_v2.py \
  --config reference_config.yaml \
  --data_path https://huggingface.co/datasets/gretelai/synthetic_text_to_sql \
  --output_dir hf://username/nemotron-nano-v2-lora-adapter

# Export adapter
python export_adapter.py --checkpoint /path/to/checkpoint
```

**Describe alternatives you've considered**

* Manually tuning hyperparameters for each use case (current state - time-consuming and requires expertise)
* Using in-context learning instead of fine-tuning (simpler but doesn't capture task-specific improvements as effectively)

**Additional context**

A reference recipe would significantly lower the barrier to entry for fine-tuning Nemotron models and provide a reproducible baseline for the community. The recipe should demonstrate that fine-tuned models maintain their core capabilities (such as generating structured reasoning outputs) while improving on the target task.
Suggested benchmark path: Text-to-SQL using data synthetically generated via NeMo Data Designer to the BIRD benchmark could serve as a strong reference case, as it combines reasoning with structured output generation.

**Comment by snowmanwwg** [comment]
Thanks for the ticket. A few things
1. We will add support for nano, and all nemotron models in the Nemo Pytorch backend, i.e. Nemo Automodel (https://github.com/NVIDIA-NeMo/Automodel). That will automatically take care off the issues on lora export etc.
2. We will add recipes for Nemotron models (incl nano-v2) in the nemotron repo (to be public) soon

**Comment by pzelasko** [comment]
Closing this issue as "won't fix".
The following collections have been moved to separate repos in https://github.com/NVIDIA-NeMo organization: avlm, llm, multimodal, multimodal-autoregressive, vlm, speechlm, diffusion
Please move the discussion to the related repo if the issue persists.
